packages/lab/src/date-picker/DatePicker.tsx

```diff
@@ -0,0 +1,167 @@
+import { clsx } from "clsx";
+import { ComponentPropsWithoutRef, forwardRef, useRef, useState } from "react";
+
+import {
+  Button,
+  makePrefixer,
+  useControlled,
+  useFloatingUI,
+  useForkRef,
+  useFormFieldProps,
+} from "@salt-ds/core";
+import { DatePickerContext } from "./DatePickerContext";
+import { DatePickerPanel } from "./DatePickerPanel";
+import { flip, size, useDismiss, useInteractions } from "@floating-ui/react";
+import { DateInput } from "../date-input";
+import { DateValue } from "@internationalized/date";
+import { CalendarIcon } from "@salt-ds/icons";
+import { CalendarProps } from "../calendar";
+
+const withBaseName = makePrefixer("saltDatePicker");
+
+export interface DatePickerProps
+  extends Omit<ComponentPropsWithoutRef<"div">, "defaultValue">,
+    Pick<
+      ComponentPropsWithoutRef<"input">,
+      "disabled" | "value" | "defaultValue" | "placeholder"
+    > {
+  selectionVariant?: "default" | "range";
+  disabled?: boolean;
+  startDate?: DateValue;
+  defaultStartDate?: DateValue;
+  endDate?: DateValue;
+  defaultEndDate?: DateValue;
+  CalendarProps?: Partial<CalendarProps> | undefined;
+  /**
+   * Function to format the input value.
+   */
+  dateFormatter?: (input: string) => string;
+}
+
+export const DatePicker = forwardRef<HTMLDivElement, DatePickerProps>(
+  function DatePicker(
+    {
+      selectionVariant = "default",
+      disabled = false,
+      placeholder = "dd mmm yyyy",
+      startDate: startDateProp,
+      endDate: endDateProp,
+      defaultStartDate,
+      defaultEndDate,
+      dateFormatter,
+      CalendarProps,
+      ...rest
+    },
+    ref
+  ) {
+    const [open, setOpen] = useState<boolean>(false);
+
+    const [startDate, setStartDate] = useControlled({
+      controlled: startDateProp,
+      default: defaultStartDate,
+      name: "StartDate",
+      state: "value",
+    });
+    const [endDate, setEndDate] = useControlled({
+      controlled: endDateProp,
+      default: defaultEndDate,
+      name: "EndDateInput",
+      state: "value",
+    });
+
+    const { x, y, strategy, elements, floating, reference, context } =
+      useFloatingUI({
+        open: open,
+        onOpenChange: setOpen,
+        placement: "bottom-start",
+        middleware: [
+          size({
+            apply({ rects, elements, availableHeight }) {
+              Object.assign(elements.floating.style, {
+                minWidth: `${rects.reference.width}px`,
+                maxHeight: `max(calc((var(--salt-size-base) + var(--salt-spacing-100)) * 5), calc(${availableHeight}px - var(--salt-spacing-100)))`,
```

Comment:
We can simplify this, this was for the items in a list but since DatePicker doesn't have them it's fine. We can probably just remove this for now.